src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -329,8 +328,17 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
 
                 services.AddSingleton<IFileLoggingStatusManager, FileLoggingStatusManager>();
 
-                if (!applicationHostOptions.HasParentScope)
+                if (applicationHostOptions.HasParentScope)
+                {
+                    // Forward th host LanguageWorkerOptions to the Job Host.
```

Comment:
super nit: can we clarify this comment a little more in terms of where the options are being forwarded from i.e. 'Forward the LanguageWorkerOptions from the parent (WebHost?) to the ScriptHost.'

Also just a refresher for myself: "HasParentScope" is used to check if this instance is running inside the WebHost or not correct? If there is no parent, does this mean that this instance is the parent? Or not necessarily?

And does not having a parent mean that it is the WebHost or is that still a ScriptHost but without a parent? Does that make it rogue if so?

Reply:
Flagging the current PR as a draft as @kshyju is helping iterate on the changes.

Will update the comment for clarity, but to answer your question: `HasParentScope` has the behavior you mentioned. If this is `false`, it signals that we're not running withing a parent container, and services that would otherwise be registered at the parent scope level must be registered within the JobHost container, otherwise they'd be missing. This is only ever the case when running outside of the WebHost, and the only scenario we maintain where that is the case are tests.

Reply:
Thank you!